.github/workflows/release.yml

```diff
@@ -0,0 +1,106 @@
+          # Fetch all tags
+          git fetch --tags
+          
+          # Get count of existing tags that match our base version
+          TAG_COUNT=$(git tag -l "v${BASE_VERSION}.*" | wc -l)
+          
+          # Use tag count as patch version
+          PATCH_VERSION="$TAG_COUNT"
+          
+          # Set the full version
+          FULL_VERSION="v${BASE_VERSION}.${PATCH_VERSION}"
+          echo "new_tag=${FULL_VERSION}" >> $GITHUB_OUTPUT
+          
+          # Generate changelog
+          # Get all version tags and sort them, excluding the current version and anything newer
+          PREV_TAG=$(git tag -l "v*.*.*" | sort -V | while read tag; do
+            # Only keep tags that are strictly less than our new version
+            if [ "$(echo -e "${FULL_VERSION}\n${tag}" | sort -V | head -n1)" = "${tag}" ] && [ "${tag}" != "${FULL_VERSION}" ]; then
+              echo "$tag"
+            fi
+          done | tail -n1 || echo "")
+          
+          if [ -z "$PREV_TAG" ]; then
+            # If no previous tag, use all commits
+            CHANGELOG=$(git log --pretty=format:"* %s" || echo "Initial release")
+          else
+            # Get commits since last tag
+            CHANGELOG=$(git log --pretty=format:"* %s" ${PREV_TAG}..HEAD || echo "No changes")
+          fi
+          {
+            echo "changelog<<EOF"
+            echo "## What's Changed"
+            echo "${CHANGELOG}"
+            echo "## How to use"
+            echo "This section is still a WIP, I will update it once I am able to better test out the execution of the files we are generating."
+            echo "### Windows"
+            echo "After downloading the windows binary, you should just be able to run the executable and after allowing it to run, you should be able to use it."
+            echo "### MacOS"
+            echo "After downloading the macos binary, you will need to run the `UART-CL By TheCod3r` command once to have macos gatekeeper block it, from there you can open your settings app, then scroll down to Security section and find the blocked app. Note a seperate lib for serial connections for macos is required which is why `libSystem.IO.Ports.Native.dylib` exists, it too will be blocked and will require unblocking."
+            echo "### Linux"
+            echo "Theoretically this works on all x86_64 linux distros, if you have any issues please let me know."
+            echo ""
+            echo "EOF"
+          } >> "$GITHUB_OUTPUT"
+
+      - name: Create Release
+        uses: softprops/action-gh-release@v1
+        with:
+          tag_name: ${{ steps.version.outputs.new_tag }}
+          name: Release ${{ steps.version.outputs.new_tag }}
+          body: ${{ steps.version.outputs.changelog }}
+          files: |
+            artifacts/UART-CL-linux-x64.zip
+            artifacts/UART-CL-osx-arm64.zip
+            artifacts/UART-CL-win-x64/UART-CL By TheCod3r.exe
+          draft: false
+          prerelease: false
```

Comment:
if you don't like automated full releases we can play around with using drafts OR pre-releases instead